src/Microsoft.ML.TimeSeries/STL/LocalRegression.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.ML.Internal.CpuMath.Core;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// this class is used to maintain the neighbors of a given particular point.
+    /// </summary>
+    internal class LocalRegression
+    {
+        private const double NumericalThreshold = 1.0e-10;
+        private readonly IReadOnlyList<double> _x;
+        private readonly IReadOnlyList<double> _y;
+        private int _length;
```

Comment:
nit: make `_length` `readonly`

Reply:
removed it from the member variable list and make it as a local variable in Decompose function as it is assigned as the length of the input series.